libraries/eosiolib/contracts/eosio/key_value.hpp

```diff
@@ -624,10 +617,10 @@ class kv_table {
        * @return An iterator to the object with the lowest key (by this index) in the table.
        */
       iterator begin() const {
-         uint32_t itr = internal_use_do_not_use::kv_it_create(config.db_name, config.contract_name.value, config.prefix.data(), config.prefix.size());
+         uint32_t itr = internal_use_do_not_use::kv_it_create(tbl->db_name, contract_name.value, prefix.data(), prefix.size());
          int32_t itr_stat = internal_use_do_not_use::kv_it_lower_bound(itr, "", 0);
 
-         return {itr, static_cast<typename iterator::status>(itr_stat), config};
+         return {itr, static_cast<typename iterator::status>(itr_stat), reinterpret_cast<const kv_index*>(this)};
```

Comment:
You don't need a cast for a derived to base conversion.  Using reinterpret_cast is undefined behavior in this case.

Reply:
When I added this, it failed to find the correct `iterator` constructor otherwise. Removing it now works though.

Reply:
If you need a cast, the correct cast for this scenario is static_cast.